test/main.test.ts

```diff
@@ -0,0 +1,54 @@
+import { expect, test } from "vitest";
+import { CurlGenerator } from "../src/main";
+import * as params from "../example/parameters";
+
+Object.keys(params).forEach((key) => {
+  test(key, () => {
+    const result = CurlGenerator(params[key], params[key].options);
+    expect(result).toMatchSnapshot();
```

Comment:
nice, I like the approach of having the results as snapshots, make things cleaner